bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinAcUnitHandler.java

```diff
@@ -191,16 +268,57 @@ private void pollStatus() throws IOException {
         }
     }
 
+    private void pollAirbaseStatus() throws IOException {
+        AirbaseControlInfo controlInfo = webTargets.getAirbaseControlInfo();
+        updateStatus(ThingStatus.ONLINE);
+        if (controlInfo != null) {
+            updateState(DaikinBindingConstants.CHANNEL_AC_POWER, controlInfo.power ? OnOffType.ON : OnOffType.OFF);
+            updateTemperatureChannel(DaikinBindingConstants.CHANNEL_AC_TEMP, controlInfo.temp);
+            updateState(DaikinBindingConstants.CHANNEL_AC_MODE, new StringType(controlInfo.mode.name()));
+            updateState(DaikinBindingConstants.CHANNEL_AIRBASE_AC_FAN_SPEED, new StringType(controlInfo.fanSpeed.name()));
+        }
+
+        SensorInfo sensorInfo = webTargets.getAirbaseSensorInfo();
+        if (sensorInfo != null) {
+            updateTemperatureChannel(DaikinBindingConstants.CHANNEL_INDOOR_TEMP, sensorInfo.indoortemp);
+
+            updateTemperatureChannel(DaikinBindingConstants.CHANNEL_OUTDOOR_TEMP, sensorInfo.outdoortemp);
+        }
+        AirbaseModelInfo modelInfo = webTargets.getAirbaseModelInfo();
+        AirbaseZoneInfo zoneInfo = webTargets.getAirbaseZoneInfo();
+        if (zoneInfo != null) {
+            updateState(DaikinBindingConstants.CHANNEL_AIRBASE_AC_ZONE1, zoneInfo.zone[1] ? OnOffType.ON : OnOffType.OFF);
```

Comment:
You can replace all these state changes with a single statement:
```
IntStream.range(0, zoneInfo.zone.length).forEach(
    idx -> updateState(DaikinBindingConstants.CHANNEL_AIRBASE_AC_ZONE + idx, OnOffType.from(zoneInfo.zone[idx])));
```

Reply:
Fixed in 4cafb310da